src/app/components/transcript/transcript.component.ts

```diff
@@ -77,16 +77,20 @@ export class TranscriptComponent implements OnInit, AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
-    combineLatest([this.conversation.highlight$, interval(100)]).subscribe(([highlight]) => {
+    combineLatest([this.conversation.highlight$, this.conversation.latestOngoingSubject, interval(100)]).subscribe(([highlight, latestOngoing]) => {
       this.currentHighlight = highlight;
 
-      if (!this.container?.nativeElement || !highlight) {
+      if (!this.container?.nativeElement) {
         return;
       }
 
+      const id = highlight ? highlight.id : latestOngoing ? latestOngoing.id : null;
+      if(!id) {
+        return;
+      }
 
       const part = this.container.nativeElement.querySelector(
-        `[data-part-id="${highlight.id}"]`,
+        `[data-part-id="${id}"]`,
       );
       if (part) {
         part.scrollIntoView({ behavior: 'smooth', block: 'center' });
```

Comment:
I think if we already scrolled an element into a viewport, we should *not* scroll for that element anymore afterwards. otherwise this can lead to very awkward scrolling behaviour where the user wants to scroll somewhere, but the program keeps scrolling back to the same view.
(e.g. if the user wants to scroll up to read an older part of the conversation, then the program should not "pull down" every 100 ms, or at least not until something new and noteworthy arrives)

Reply:
agreed, but haven't found a good solution yet. problem is that the subject is triggered before the element is rendered and its unclear when its rendered. one solution could be on the ngAfterViewInit on an message element in the transcript, but didnt want to change it too drastic before the event.

the current change just adds scrolling to the ongoing live text.